x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/EngineIndexService.java

```diff
@@ -205,6 +265,15 @@ public void putEngine(Engine engine, ActionListener<IndexResponse> listener) {
         }
     }
 
+    private void validateIndices(Engine engine) {
+        for (String index : engine.indices()) {
+            final IndexMetadata indexMetadata = clusterService.state().metadata().index(index);
+            if (indexMetadata == null) {
```

Comment:
yep, I thought about that too. I think this way it is consistent with other ES behaviour (for example, searching on multiple indices) and we simplify this code as well.

Reply:
Are we accepting index patterns? If so, this won't work with this validation. I also wonder why we need this validation since the alias action checks the existence of the indices already.

Reply:
> Are we accepting index patterns?

I was thinking that no, until we added IndicesRequest.Replaceable to the Request. Wouldn't that take care of index patterns and replace them on the request?

>  I also wonder why we need this validation since the alias action checks the existence of the indices already

That returned a not found response from the alias, and was a bit confusing since we were returning a 404 directly. I could capture that result from the alias operation and issue a 400 response to make clear we were trying to use a non-existing index.